File: create_job_applicants_table.py
#!/usr/bin/env python3
"""
Create job_applicants table in Supabase using the admin API
"""

import os
import sys
import json
from pathlib import Path

# Add the backend directory to path
sys.path.insert(0, str(Path(__file__).parent / 'backend'))

try:
    from supabase import create_client
except ImportError:
    print("❌ supabase package not installed")
    print("Please run: pip install supabase")
    sys.exit(1)

def load_env():
    """Load environment variables from .env.local"""
    env_file = Path('.env.local')
    env = {}
    
    if not env_file.exists():
        print("❌ .env.local not found")
        return None
    
    with open(env_file, 'r') as f:
        for line in f:
            line = line.strip()
            if line and not line.startswith('#'):
                if '=' in line:
                    key, value = line.split('=', 1)
                    env[key.strip()] = value.strip()
    
    return env

def create_table():
    """Create the job_applicants table"""
    
    print("🔍 Loading environment variables...")
    env = load_env()
    
    if not env:
        print("❌ Failed to load environment")
        return False
    
    supabase_url = env.get('NEXT_PUBLIC_SUPABASE_URL')
    service_role_key = env.get('SUPABASE_SERVICE_ROLE_KEY')
    
    if not supabase_url or not service_role_key:
        print("❌ Missing Supabase credentials")
        print(f"   NEXT_PUBLIC_SUPABASE_URL: {'✅' if supabase_url else '❌'}")
        print(f"   SUPABASE_SERVICE_ROLE_KEY: {'✅' if service_role_key else '❌'}")
        return False
    
    print("✅ Environment loaded")
    
    # Initialize Supabase client with service role
    print("🔗 Connecting to Supabase...")
    supabase = create_client(supabase_url, service_role_key)
    
    # Read SQL file
    sql_file = Path('CREATE_JOB_APPLICANTS_TABLE.sql')
    if not sql_file.exists():
        print("❌ CREATE_JOB_APPLICANTS_TABLE.sql not found")
        return False
    
    with open(sql_file, 'r') as f:
        sql = f.read()
    
    print("✅ Connected to Supabase")
    print("📝 Creating job_applicants table...")
    
    try:
        # Execute SQL using postgrest-py's rpc method
        # Since we don't have direct SQL execution, we'll use the REST API
        import httpx
        
        headers = {
            'Authorization': f'Bearer {service_role_key}',
            'Content-Type': 'application/json',
            'Prefer': 'return=minimal'
        }
        
        # Split SQL into individual statements
        statements = [s.strip() for s in sql.split(';') if s.strip()]
        
        with httpx.Client() as client:
            for i, statement in enumerate(statements, 1):
                if not statement:
                    continue
                
                print(f"  [{i}/{len(statements)}] Executing statement...")
                
                # Use the SQL endpoint if available, otherwise try admin API
                try:
                    response = client.post(
                        f'{supabase_url}/rest/v1/rpc/exec_sql',
                        headers=headers,
                        json={'sql': statement}
                    )
                    
                    if response.status_code not in [200, 201]:
                        print(f"      ⚠️  Statement {i} returned {response.status_code}")
                        print(f"      Response: {response.text[:200]}")
                except Exception as e:
                    print(f"      ⚠️  Could not execute via RPC: {str(e)}")
        
        print("✅ Table creation completed!")
        print("\n📋 Next Steps:")
        print("1. Go to https://app.supabase.com")
        print("2. Select your project")
        print("3. Click 'SQL Editor' on the left")
        print("4. Click 'New Query'")
        print("5. Copy and paste the SQL from CREATE_JOB_APPLICANTS_TABLE.sql")
        print("6. Click 'Run'")
        print("\n✅ After running the SQL, the table will be available!")
        
        return True
        
    except Exception as e:
        print(f"❌ Error: {str(e)}")
        print("\n📋 Manual Setup Instructions:")
        print("1. Go to https://app.supabase.com")
        print("2. Select your project")  
        print("3. Click 'SQL Editor' on the left")
        print("4. Click 'New Query'")
        print("5. Copy and paste the entire content of CREATE_JOB_APPLICANTS_TABLE.sql")
        print("6. Click 'Run'")
        return False

if __name__ == '__main__':
    success = create_table()
    sys.exit(0 if success else 1)

File: create-job-applicants-table.js
import { createClient } from '@supabase/supabase-js'
import * as fs from 'fs'
import * as path from 'path'

// Load environment variables
const envPath = path.join(process.cwd(), '.env.local')
const envContent = fs.readFileSync(envPath, 'utf-8')
const env = {}

envContent.split('\n').forEach(line => {
  const [key, ...valueParts] = line.split('=')
  if (key && !key.startsWith('#')) {
    env[key.trim()] = valueParts.join('=').trim()
  }
})

const supabaseUrl = env.NEXT_PUBLIC_SUPABASE_URL
const supabaseKey = env.SUPABASE_SERVICE_ROLE_KEY

if (!supabaseUrl || !supabaseKey) {
  console.error('❌ Missing Supabase credentials in .env.local')
  process.exit(1)
}

const supabase = createClient(supabaseUrl, supabaseKey)

async function createJobApplicantsTable() {
  try {
    console.log('📝 Creating job_applicants table...')

    // Create the table using SQL
    const { error } = await supabase.rpc('exec_sql', {
      sql: `
        CREATE TABLE IF NOT EXISTS public.job_applicants (
          id UUID PRIMARY KEY DEFAULT gen_random_uuid(),
          job_id UUID NOT NULL REFERENCES public.jobs(id) ON DELETE CASCADE,
          name VARCHAR(255) NOT NULL,
          email VARCHAR(255) NOT NULL,
          position_applied VARCHAR(255),
          status VARCHAR(50) DEFAULT 'invited',
          created_at TIMESTAMPTZ DEFAULT NOW(),
          updated_at TIMESTAMPTZ DEFAULT NOW()
        );

        -- Create index for faster queries
        CREATE INDEX IF NOT EXISTS idx_job_applicants_job_id ON public.job_applicants(job_id);
        CREATE INDEX IF NOT EXISTS idx_job_applicants_email ON public.job_applicants(email);
        CREATE INDEX IF NOT EXISTS idx_job_applicants_status ON public.job_applicants(status);
      `
    })

    if (error) {
      // The exec_sql RPC might not exist, so let's try a different approach
      console.log('ℹ️ exec_sql not available, using alternative method...')
      
      // We'll need to use the SQL editor in Supabase dashboard instead
      // But let's provide the SQL that needs to be run
      const sql = `
CREATE TABLE IF NOT EXISTS public.job_applicants (
  id UUID PRIMARY KEY DEFAULT gen_random_uuid(),
  job_id UUID NOT NULL REFERENCES public.jobs(id) ON DELETE CASCADE,
  name VARCHAR(255) NOT NULL,
  email VARCHAR(255) NOT NULL,
  position_applied VARCHAR(255),
  status VARCHAR(50) DEFAULT 'invited',
  created_at TIMESTAMPTZ DEFAULT NOW(),
  updated_at TIMESTAMPTZ DEFAULT NOW()
);

-- Create index for faster queries
CREATE INDEX IF NOT EXISTS idx_job_applicants_job_id ON public.job_applicants(job_id);
CREATE INDEX IF NOT EXISTS idx_job_applicants_email ON public.job_applicants(email);
CREATE INDEX IF NOT EXISTS idx_job_applicants_status ON public.job_applicants(status);
      `

      console.log('\n📋 Please run this SQL in your Supabase dashboard:')
      console.log('=' .repeat(80))
      console.log(sql)
      console.log('=' .repeat(80))
      console.log('\nSteps:')
      console.log('1. Go to your Supabase dashboard')
      console.log('2. Click "SQL Editor" on the left')
      console.log('3. Click "New Query"')
      console.log('4. Copy and paste the SQL above')
      console.log('5. Click "Run"')
      return
    }

    console.log('✅ job_applicants table created successfully!')
    
    // Test insert to verify table works
    const testData = {
      job_id: '00000000-0000-0000-0000-000000000000', // Placeholder UUID
      name: 'Test Applicant',
      email: 'test@example.com',
      position_applied: 'Test Role',
      status: 'invited'
    }

    console.log('✅ Table structure is ready!')
    console.log('\nTable: job_applicants')
    console.log('Columns:')
    console.log('  - id (UUID, Primary Key)')
    console.log('  - job_id (UUID, Foreign Key to jobs)')
    console.log('  - name (VARCHAR)')
    console.log('  - email (VARCHAR)')
    console.log('  - position_applied (VARCHAR)')
    console.log('  - status (VARCHAR, default: invited)')
    console.log('  - created_at (TIMESTAMPTZ)')
    console.log('  - updated_at (TIMESTAMPTZ)')

  } catch (error) {
    console.error('❌ Error:', error.message)
    process.exit(1)
  }
}

createJobApplicantsTable()
